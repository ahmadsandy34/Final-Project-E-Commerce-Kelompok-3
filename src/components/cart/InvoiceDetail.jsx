import React from "react";

const InvoiceDetail = () => {
  return (
    <div>
      <div className="flex justify-between border-b-2 border-gray-400 py-3">
        <p>Subtotal</p>
        <p>$1750</p>
      </div>
      <div className="flex justify-between border-b-2 border-gray-400 py-3">
        <p>Shipping</p>
        <p>Free</p>
      </div>
      <div className="flex justify-between py-3">
        <p>Total</p>
        <p className="font-semibold">$1750</p>
      </div>
    </div>
  );
};

export default InvoiceDetail;
