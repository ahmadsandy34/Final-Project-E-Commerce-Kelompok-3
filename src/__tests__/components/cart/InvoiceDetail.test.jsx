import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import InvoiceDetail from "../../../components/cart/InvoiceDetail";

describe("InvoiceDetail Component", () => {
  test("renders shipping as Free", () => {
    render(<InvoiceDetail total={50} />);
    expect(screen.getByText("Free")).toBeInTheDocument();
    expect(screen.getByText("Shipping")).toBeInTheDocument();
  });

  test("renders total with correct amount", () => {
    render(<InvoiceDetail total={75} />);
    const totalElements = screen.getAllByText("$75");
    expect(totalElements).toHaveLength(2);
    expect(screen.getByText("Total")).toBeInTheDocument();
  });

  test("renders with zero total", () => {
    render(<InvoiceDetail total={0} />);
    const zeroElements = screen.getAllByText("$0");
    expect(zeroElements).toHaveLength(2);
  });

  test("matches snapshot", () => {
    const { container } = render(<InvoiceDetail total={150} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
