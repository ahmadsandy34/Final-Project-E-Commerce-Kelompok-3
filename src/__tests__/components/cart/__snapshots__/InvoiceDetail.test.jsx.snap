// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceDetail Component matches snapshot 1`] = `
<div>
  <div
    class="flex justify-between border-b-2 border-gray-400 py-3"
  >
    <p>
      Subtotal
    </p>
    <p>
      $
      150
    </p>
  </div>
  <div
    class="flex justify-between border-b-2 border-gray-400 py-3"
  >
    <p>
      Shipping
    </p>
    <p>
      Free
    </p>
  </div>
  <div
    class="flex justify-between py-3"
  >
    <p>
      Total
    </p>
    <p
      class="font-semibold"
    >
      $
      150
    </p>
  </div>
</div>
`;
